### DISCUSSION TITLE
cannot compile latex file

### DISCUSSION THREAD

**Discussion by mrhabibbi** [James-Yu/LaTeX-Workshop#3223] [Q&A]
on vscode macOS trying to compile latex file but produces errors:

- Recipe terminated with fatal error: spawn latexmk ENOENT.
- chktex could not be found.

any help is appreciated

**Comment by James-Yu** [comment]
Check you system PATH please. In fact not related to this extension.

**Reply by James-Yu** [reply]
You can `echo $PATH` in Terminal and see if your LaTeX installation is there.

**Reply by mrhabibbi** [reply]
> You can `echo $PATH` in Terminal and see if your LaTeX installation is there.

mac:~ user$ echo $PATH
/Library/Frameworks/Python.framework/Versions/3.10/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:/opt/X11/bin:~/.dotnet/tools:/Library/Apple/usr/bin:/Library/Frameworks/Python.framework/Versions/3.10/bin:/opt/metasploit-framework/bin:/opt/metasploit-framework/bin:/opt/metasploit-framework/bin

**Reply by mrhabibbi** [reply]
not sure how i can check if its installed

**Reply by James-Yu** [reply]
Obviously it is not. Please refer to your LaTeX distribution website for instruction.

If you are not sure, or did not install, I personally recommend you to check https://tug.org/mactex/.

**Reply by vpozdnyakov** [reply]
@James-Yu it is not obvious how to find the instructions in https://tug.org/mactex/ which lines should be added to PATH in a case if MacTex is installed

**Comment by bitecc** [comment]
Same question and a little confused. First to say, I use wsl remote, and the `latexmk` command is config in `~.bashrc` file. In vscode integrated shell I can surely run`latexmk`. But using this extension to build tex file, I got this message
> Does the executable exist? $PATH: /home/joshua/.vscode-server/bin/8dfae7a5cd50421d10cd99cb873990460525a898/bin/remote-cli:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/lib/wsl/lib

It vividly showed that this extension only use my system path file but not my personal path file. That's why the `latexmk` command didn't work, given that this command is not in /usr/bin or /usr/local/bin. I know it is the matter of vscode, but I am a little confused about how this extension handles the $PATH variable. Why does it only get my system $PATH?

**Reply by bitecc** [reply]
By the way I sloved this problem by linking `latexmk` to /usr/local/bin/ as expected. I have used this awsome extension for alomost three years. This question never happened until I updated my vscode last night. I think it cound be possible that vscode changed the APIs for developers so it provided the system $PATH but not my personal $PATH file (`.bashrc`) to these extensions for some safty considerstion, unlike the vscode integrated shell, where I can use my personal $PATH config.

**Reply by Dawntown** [reply]
I am in the same situation as you, but I tend to keep the PATH clean. Do you find other solutions instead of linking latex to system $PATH?

**Reply by harith-alsafi** [reply]
I have the exact same issue as well, I sure have my paths correct but once I updated vscode and my OS I started seeing this error, so any updates ?
